eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/sorted/mutable/SynchronizedSortedBag.java

```diff
@@ -655,4 +656,22 @@ public int detectLastIndex(Predicate<? super T> predicate)
     {
         throw new UnsupportedOperationException("asParallel() method is not supported for " + this.getClass().getSimpleName() + '.');
     }
+
+    @Override
+    public Optional<T> getFirstOptional()
+    {
+        synchronized (this.getLock())
+        {
+            return this.getDelegate().getFirstOptional();
+        }
+    }
+
+    @Override
+    public Optional<T> getLastOptional()
+    {
+        synchronized (this.getLock())
+        {
+            return this.getDelegate().getLastOptional();
+        }
+    }
```

Comment:
`AbstractSynchronizedRichIterable` does not implement `OrderedIterable` (only `RichIterable`). I added `get(First|Last)Optional` to `OrderedIterable` (as requested), while `get(First|Last)` are part of `RichIterable`. I think we would have to add another class e.g. `AbstractSynchronizedOrderedIterable` to have a central place to add `get(First|Last)Optional` for the synchronized classes.